File: Hearo.Application/Services/Blogs/BlogService.cs
using AutoMapper;
using Hearo.Application.Common.Interfaces.Persistence;
using Hearo.Application.Common.Interfaces.Services;
using Hearo.Application.Common.Models.Blogs;
using Microsoft.EntityFrameworkCore;

namespace Hearo.Application.Services.Blogs;

public class BlogService : IBlogService
{
    private readonly IApplicationDbContext _context;
    private readonly IMapper _mapper;

    public BlogService(IApplicationDbContext context, IMapper mapper)
    {
        _context = context;
        _mapper = mapper;
    }

    public async Task<List<BlogDto>> GetAllBlogs()
    {
        var blogs = await _context.Blogs.Include(b => b.Author).ToListAsync();
        return _mapper.Map<List<BlogDto>>(blogs);
    }

    public async Task<BlogDto> GetBlogBySlug(string slug)
    {
        var blog = await _context.Blogs.Include(b => b.Author).FirstOrDefaultAsync(x => x.Slug == slug);
        return _mapper.Map<BlogDto>(blog);
    }
}

File: podsphere-web/src/services/blog.service.ts
import { api } from "../lib/api";
import { BlogDto } from "../types/social";

export const blogService = {
  // Lấy tất cả bài blog
  getAllBlogs: async () => {
    const { data } = await api.get<BlogDto[]>("/blogs");
    return data;
  },

  // Lấy chi tiết một bài blog theo Slug (để làm URL đẹp) hoặc ID
  getBlogBySlug: async (slug: string) => {
    const { data } = await api.get<BlogDto>(`/blogs/${slug}`);
    return data;
  }
};